client/my-sites/theme/main.jsx

```diff
@@ -1429,9 +1454,12 @@ export default connect(
 
 		const queryArgs = getCurrentQueryArguments( state );
 
+		const isThemeSelectedTask = window.location.hash === HOSTING_THEME_SELCETED_HASH;
```

Comment:
We don't want to check this in the `connect()` function; the `mapStateToProps` function gets called every render, which means when the hash changes so does the `isThemeSelectedTask` prop. We want to check the hash when the component is first initialised, just like we do in the site monitoring component with `useRef`